controller_manager/scripts/spawner.py

```diff
@@ -0,0 +1,138 @@
+        '-p', '--param-file', help='Controller param file to be loaded into controller '
+        'node before configure', required=False)
+    parser.add_argument(
+        '-u', '--unload-on-kill',
+        help='Wait until this application is interrupted and unload controller',
+        action='store_true')
+
+    command_line_args = rclpy.utilities.remove_ros_args(args=sys.argv)[1:]
+    args = parser.parse_args(command_line_args)
+    controller_name = args.controller_name
+    controller_manager_name = args.controller_manager
+    param_file = args.param_file
+    controller_type = args.controller_type
+
+    node = Node('spawner_' + controller_name)
+    try:
+
+        # Wait for controller_manager
+        timeout = node.get_clock().now() + Duration(seconds=10)
+        while node.get_clock().now() < timeout:
+            ret = subprocess.run(
+                ['ros2', 'service', 'type',
+                 '/' + controller_manager_name + '/load_and_start_controller'],
+                stdout=subprocess.DEVNULL,
+                stderr=subprocess.DEVNULL)
+            if ret.returncode == 0:
+                break
+            node.get_logger().info(
+                'Waiting for {} services'.format(controller_manager_name),
+                throttle_duration_sec=2)
+            time.sleep(0.2)
+
+        if controller_type:
+            ret = subprocess.run(['ros2', 'param', 'set', controller_manager_name,
+                                  controller_name + '.type', controller_type])
+
+        if is_controller_loaded(controller_manager_name, controller_name):
+            node.get_logger().info('Controller already loaded, skipping load_controller')
+        else:
+            ret = subprocess.run(['ros2', 'control', 'load_controller', controller_name,
+                                  '--controller-manager', controller_manager_name])
+            if ret.returncode != 0:
+                # Error message printed by ros2 control
+                return ret.returncode
+            node.get_logger().info('Loaded ' + controller_name)
+
+        if param_file:
+            ret = subprocess.run(['ros2', 'param', 'load', controller_name,
+                                  param_file])
+            if ret.returncode != 0:
+                # Error message printed by ros2 param
+                return ret.returncode
+            node.get_logger().info('Loaded ' + param_file + ' into ' + controller_name)
+
+        ret = subprocess.run(['ros2', 'control', 'configure_start_controller', controller_name,
+                              '--controller-manager', controller_manager_name])
+        if ret.returncode != 0:
+            # Error message printed by ros2 control
+            return ret.returncode
+        node.get_logger().info('Configured and started ' + controller_name)
+
+        if not args.unload_on_kill:
+            return 0
+        try:
+            node.get_logger().info('Waiting until interrupt to unload controllers')
```

Comment:
This could clutter the output. Maybe to throttle this to 60s or something?

Reply:
It shouldn't, the print is before the loop. Will only be printed once.